spec/requests/campaigns_spec.rb

```diff
@@ -0,0 +1,96 @@
+require 'rails_helper'
+
+describe "Campaigns", type: :request do
+  before do
+    login_as(create(:user), scope: :user)
+  end
+
+  describe "#create" do
+    context "given valid params" do
+      let(:params) do
+        { campaign: { name: "Super Campaign" } }
+      end
+
+      it "creates a new campaign" do
+        expect {
+          post "/campaigns", params
+        }.to change(Campaign, :count).by(1)
+      end
+
+      it "redirects to /campaigns" do
+        post "/campaigns", params
+        expect(response).to redirect_to "/campaigns"
+      end
+
+      it "publishes the event" do
+        expect(ChampaignQueue).to receive(:push) do |hash|
```

Comment:
Interesting way of checking arguments. Would normally expect to see the `hash_including` matcher which I think is a little easier to follow:

```ruby
expect(ChampaignQueue).to receive(:push).with(hash_including({
  type: 'Super Campaign',
  campaign_id: be_a(Integer)
})
```

What do you think?